### PR TITLE
Isaiah fowler

### PR DESCRIPTION
**Pull request by lovesinatra**
# PR Description
​
- Fixed bug inside the notes router​
- Added "shareable" flag to the notes model
- Created route inside families router that sends back all notes related to that family
- Guests are restricted to only viewing shareable notes
​
## Type of Change
​
Types of Change:
- [] New feature
- [] Refactoring / Bug fix 
​

## Testing
​
- [] No​
​
## Completeness
Is this feature complete
- [] Feature Complete

​

### CODE REVIEW

File: api/families/familiesModel.js [hunk 1]
```diff
@@ -39,6 +39,19 @@ const findAllFamilyMembersById = async (id) => {
...
+    .where({ family_id: id })
+    .returning('*')
+    .modify((qb) => {
+      // Only send back the shareable notes if the user is a guest
+
+      if (role == 'guest') {
+        qb.where({ shareable: true });
+      }
+    });
+};
```
Comment (thereactgirl) [inline api/families/familiesModel.js:53]: Awesome thanks for adding the restriction here of notes that guests should not be able to view! I was wondering how to do this and you did it super clean and concise. BRAVO

File: api/families/familiesModel.js [hunk 2]
```diff
@@ -39,6 +39,19 @@ const findAllFamilyMembersById = async (id) => {
   return db('members').where({ family_id: id }).returning('*');
 };
 
+const findAllNotesByFamilyId = async (id, role) => {
```
Comment (thereactgirl) [inline api/families/familiesModel.js:42]: Why is there an `async` keyword but no `await`?

File: api/families/familiesRouter.js [hunk 3]
```diff
@@ -44,6 +44,19 @@ router.get('/:id/members', function (req, res) {
     });
 });
 
+router.get('/:id/notes', authRequired, function (req, res) {
+  const { role } = req.user;
```
Comment (thereactgirl) [inline api/families/familiesRouter.js:48]: Nice, I didn't think of putting a notes endpoint by family but this is def helpful!

File: data/migrations/20201209161927_notes.js [hunk 4]
```diff
@@ -23,8 +23,10 @@ exports.up = function (knex) {
     tbl.string('subject').notNullable();
 
     tbl.string('content').notNullable();
+    
+    tbl.boolean('shareable').notNullable().defaultTo(true)
 
-    tbl.date('date');
+    tbl.date('date').defaultTo(knex.fn.now())
```
Comment (thereactgirl) [inline data/migrations/20201209161927_notes.js:29]: Love this. I didn't know Knex had this functionality.

File: api/notes/notesRouter.js [hunk 5]
```diff
@@ -6,7 +6,9 @@ const router = express.Router();
 
 // checkRole.grantAccess('readAny', 'notes'),
 router.get('/', authRequired, function (req, res) {
-  Notes.findAll()
+  const queries = { ...req.query };
```
Comment (thereactgirl) [inline api/notes/notesRouter.js:7]: Does `req.query` return notes based on the URL or request body or both?

File: api/notes/notesRouter.js [hunk 6]
```diff
@@ -18,8 +20,8 @@ router.get('/', authRequired, function (req, res) {
 
 // checkRole.grantAccess('readOwn', 'notes'),
```
Comment (thereactgirl) [inline api/notes/notesRouter.js]: Can we remove this commented out code? If this is documentation it should be clear

File: api/notes/notesRouter.js [hunk 7]
```diff
@@ -18,8 +20,8 @@ router.get('/', authRequired, function (req, res) {
...
 router.get('/:id', authRequired, function (req, res) {
-  const family_id = String(req.params.id);
-  Notes.findById(family_id)
```
Comment (thereactgirl) [inline api/notes/notesRouter.js:22]: NICE CATCH thanks for fixing this!

File: api/notes/notesRouter.js [hunk 8]
```diff
@@ -91,12 +93,8 @@ router.put('/:id', authRequired, (req, res) => {
...
   const id = req.params.id;
   try {
-    Notes.findById(id).then((notes) => {
-      Notes.remove(notes['family_id']).then(() => {
-        res
-          .status(200)
-          .json({ message: `notes '${id}' was deleted.`, notes: notes });
-      });
+    Notes.findByIdAndRemove(id).then(() => {
+      res.status(204).json({ message: 'deleted' });
```
Comment (thereactgirl) [inline api/notes/notesRouter.js:91]: Beautiful refactoring.

**Review by thereactgirl — CHANGES_REQUESTED** [review]
Awesome job! Please review my comments about the commented out code and I will approve!

File: api/families/familiesModel.js (hunk 2 above)
Comment (lovesinatra) [reply api/families/familiesModel.js:42]: whoops nice catch. We can remove that

Comment (thereactgirl) [reply api/families/familiesModel.js:42]: sweet!

**Comment by abdimohamud** [comment]
looks great to me! will make sure to add new endpoint to API docs, but if we dont need to change the seeds for the date fix, then its good to migrate and re-deploy the db after the changes!

File: api/notes/notesRouter.js (hunk 6 above)
Comment (lovesinatra) [reply api/notes/notesRouter.js]: Sure thing

**Review by thereactgirl — APPROVED** [review]

**Review by abdimohamud — APPROVED** [review]

**Pull request merged**